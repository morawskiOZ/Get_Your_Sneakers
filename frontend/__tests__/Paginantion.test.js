import { mount } from "enzyme"
import Pagination, { PAGINATION_QUERY } from "../components/Pagination"
import wait from "waait"
import toJSON from "enzyme-to-json"
import Router from "next/router"
import { MockedProvider } from "react-apollo/test-utils"

Router.router = {
  push() {},
  prefetch() {}
}

function makeMockFor(length) {
  return [
    {
      request: { query: PAGINATION_QUERY },
      result: {
        data: {
          itemsConnection: {
            __typename: "aggregate",
            aggregate: {
              __typename: "count",
              count: length
            }
          }
        }
      }
    }
  ]
}

describe("<Pagination />", () => {
  it("displays a loading message", () => {
    const wrapper = mount(
      <MockedProvider mocks={makeMockFor(1)}>
        <Pagination page={1} />
      </MockedProvider>
    )
    expect(wrapper.text()).toContain("Loading")
  })

  it("renders pagination fro 18 items", async () => {
    const wrapper = mount(
      <MockedProvider mocks={makeMockFor(18)}>
        <Pagination page={1} />
      </MockedProvider>
    )
    await wait()
    wrapper.update()
    const pagination = wrapper.find("div[data-test='pagination']")
    expect(toJSON(pagination)).toMatchSnapshot()
  })
})
