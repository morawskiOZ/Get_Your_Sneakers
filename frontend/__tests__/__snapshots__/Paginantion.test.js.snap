// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders pagination fro 18 items 1`] = `
<div
  className="PaginationStyles-aduuar-0 ewJsCc"
  data-test="pagination"
>
  <SideEffect(Head)>
    <Head />
  </SideEffect(Head)>
  <Link
    href={
      Object {
        "pathname": "items",
        "query": Object {
          "page": 0,
        },
      }
    }
    prefetch={true}
  >
    <a
      aria-disabled={true}
      className="prev"
      href="items?page=0"
      onClick={[Function]}
    >
      Prev
    </a>
  </Link>
  <p>
    Page 
    1
     of 
    5
  </p>
  <p>
    18
     Items total
  </p>
  <Link
    href={
      Object {
        "pathname": "items",
        "query": Object {
          "page": 2,
        },
      }
    }
    prefetch={true}
  >
    <a
      aria-disabled={false}
      className="next"
      href="items?page=2"
      onClick={[Function]}
    >
      Next
    </a>
  </Link>
</div>
`;
